cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -153,13 +158,16 @@ func (dp *devicePlugin) scan() (dpapi.DeviceTree, error) {
 			klog.V(4).Infof("Adding %s to GPU %s", devPath, f.Name())
 			nodes = append(nodes, devSpec)
 
+			if !dp.options.enableMonitoring {
+				continue
+			}
 			klog.V(4).Infof("Adding %s to GPU %s/%s", devPath, monitorType, monitorID)
 			monitor = append(monitor, devSpec)
```

Comment:
We can get rid of negation by moving this appending to `monitor` into the added branch.

Reply:
Fixed (was remains of older version of code).